Nef_S2/include/CGAL/Nef_S2/Sphere_segment.h

```diff
@@ -37,7 +38,7 @@ Sphere_segment_rep() { ps_ = pt_ = Point(); c_ = Circle(); }
 
 Sphere_segment_rep(const Point& p1, const Point& p2,
                    bool shorter_arc=true) :
-  ps_(p1), pt_(p2), c_(Plane_3(p1,p2,Point_3(CGAL::ORIGIN)))
+  ps_(p1), pt_(p2), c_(CGAL::ORIGIN,CGAL::orthogonal_vector(p1,p2,Point_3(CGAL::ORIGIN)))
```

Comment:
This should not be a global function, but a functor of the kernel.   And, @lrineau, correct me if I am wrong, is that not the crossproduct of `Vector_3(p1)` and `Vector_3(p2)` as the third point is the origin?

Reply:
I can change this to:
```c++
c_(CGAL::ORIGIN,Plane_3(p1,p2,Point_3(CGAL::ORIGIN)).orthogonal_vector())
```
I just thought constructing a plane just for its orthogonal vector seemed unnecessary.

Reply:
But in case I am right with cross product it would even be simpler, wouldn't it?
```
c_(CGAL::ORIGIN, R_().cross_product_3_object()(Vector_3(p1), Vector_3(p2)));
```

Reply:
Ah I see. I was thinking It would still have to use the global function `CGAL::cross_product(Vector_3(p1), Vector_3(p2))` I misunderstood your suggestion.

Reply:
> correct me if I am wrong, is that not the crossproduct of

@afabri Assuming using the crossproduct is correct I think it would be:
```c++
R_().construct_cross_product_vector_3_object()(p1-CGAL::ORIGIN, p2-CGAL::ORIGIN)
```

I wonder if it's worth cleaning up this too:
```c++
 Plane_3 h(Point_3(CGAL::ORIGIN),(p1-CGAL::ORIGIN));
    c_ = Sphere_circle<R_>(Plane_3(Point_3(CGAL::ORIGIN),h.base1()));
 ```

Reply:
`Vector_3(p)` was more a symbolic writing to convey the idea  with the crossproduct function.  I am only wondering how much we will pay for    `p1 - ORIGIN`   as this constructs  nodes in the DAG of the Epeck objects. But maybe just benchmarking is better than meditating about this  kind of problem.

Reply:
Which line do you refer to  when you write "I am wondering ...." ?

Reply:
I think the assertion that cross_product == orthogonal_vector of three points when the third is the origin is correct

```c++
#include <CGAL/Exact_predicates_exact_constructions_kernel.h>
#include <CGAL/Kernel/global_functions.h>
#include <CGAL/point_generators_3.h>

using K = CGAL::Exact_predicates_exact_constructions_kernel;
using Point = CGAL::Point_3<K>;

int main(int, char*[])
{
	CGAL::Random_points_on_sphere_3<Point> rnd;

	for(int i=0; i<1000; ++i)
	{
		Point p1=*rnd++;
		Point p2=*rnd++;
		if(CGAL::cross_product(p1-CGAL::ORIGIN,p2-CGAL::ORIGIN) != CGAL::orthogonal_vector(p1,p2,Point(CGAL::ORIGIN)))
		{
			throw 1;
		}
	}
	return 0;
}
```

Reply:
> is it possible to add to the `Construct_base_vector_3` functor?
> 
> ```c++
> result_type operator()( const Vector_3& v, int index ) const { ... }
> ```

We can add any documented kernel functors  as the requirements of the traits class are not documented.

Reply:
> I think the assertion that cross_product == orthogonal_vector of three points when the third is the origin is correct

It is correct, because mathematically that is correct.

The whole situation here is because those sphere "points" should actually be vectors, or best: directions.

Reply:
> The whole situation here is because those sphere "points" should actually be vectors, or best: directions.

I agree, whilst looking at the code your suggestion became more compelling. But I think it should be a separate Issue/Pull request and if someone wants to implement it ;)

Reply:
@afabri No what I really had in mind was just this:

```c++
 Plane_3 h(CGAL::ORIGIN,(p1-CGAL::ORIGIN));
 c_ = Sphere_circle<R_>(CGAL::ORIGIN,h.base1());
```

But your other comments got me wondering...